components/datetime/src/lib.rs

```diff
@@ -84,222 +85,24 @@ pub mod options;
-            symbols: &self.symbols,
-            date_time: value,
-            locale: &self.locale,
-        }
-    }
-
-    /// `format_to_write` takes a mutable reference to anything that implements `Write` trait
-    /// and a `DateTime` value and populates the buffer with a formatted value.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// # use icu_locid::Locale;
-    /// # use icu_locid_macros::langid;
-    /// # use icu_datetime::{DateTimeFormat, DateTimeFormatOptions};
-    /// # use icu_datetime::mock::MockDateTime;
-    /// # use icu_provider::inv::InvariantDataProvider;
-    /// # let locale: Locale = langid!("en").into();
-    /// # let provider = InvariantDataProvider;
-    /// # let options = DateTimeFormatOptions::default();
-    /// let dtf = DateTimeFormat::try_new(locale, &provider, &options.into())
-    ///     .expect("Failed to create DateTimeFormat instance.");
-    ///
-    /// let date_time = MockDateTime::try_new(2020, 9, 1, 12, 34, 28)
-    ///     .expect("Failed to construct DateTime.");
-    ///
-    /// let mut buffer = String::new();
-    /// dtf.format_to_write(&mut buffer, &date_time)
-    ///     .expect("Failed to write to a buffer.");
-    ///
-    /// let _ = format!("Date: {}", buffer);
-    /// ```
-    pub fn format_to_write(
-        &self,
-        w: &mut impl std::fmt::Write,
-        value: &impl DateTimeInput,
-    ) -> std::fmt::Result {
-        write_pattern(&self.pattern, &self.symbols, value, &self.locale, w)
-            .map_err(|_| std::fmt::Error)
-    }
-
-    /// `format_to_string` takes a `DateTime` value and returns it formatted
-    /// as a string.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// # use icu_locid::Locale;
-    /// # use icu_locid_macros::langid;
-    /// # use icu_datetime::{DateTimeFormat, DateTimeFormatOptions};
-    /// # use icu_datetime::mock::MockDateTime;
-    /// # use icu_provider::inv::InvariantDataProvider;
-    /// # let locale: Locale = langid!("en").into();
-    /// # let provider = InvariantDataProvider;
-    /// # let options = DateTimeFormatOptions::default();
-    /// let dtf = DateTimeFormat::try_new(locale, &provider, &options.into())
-    ///     .expect("Failed to create DateTimeFormat instance.");
-    ///
-    /// let date_time = MockDateTime::try_new(2020, 9, 1, 12, 34, 28)
-    ///     .expect("Failed to construct DateTime.");
-    ///
-    /// let _ = dtf.format_to_string(&date_time);
-    /// ```
-    pub fn format_to_string(&self, value: &impl DateTimeInput) -> String {
-        let mut s = String::new();
-        self.format_to_write(&mut s, value)
-            .expect("Failed to write to a String.");
-        s
-    }
+#[macro_export]
+macro_rules! invalid_pattern_symbol {
+    ($category: ident, $symbol: expr, $length: expr) => {
+        panic!(
+            "Invalid pattern symbol {}({:?}) of length {}",
+            stringify!($category),
+            $symbol,
+            $length
+        )
+    };
```

Comment:
This content was moved to `datetime.rs` so that `datetime`, `timezone`, and `zoned_datetime` could all be in their own file modules.